src/raqm.c

```diff
@@ -158,6 +162,42 @@
 # define RAQM_TEST(...)
 #endif
 
+#ifdef RAQM_SHEENBIDI
+# define RAQM_PAR_ON  SBBidiTypeON
+# define RAQM_PAR_RTL SBBidiTypeR
+# define RAQM_PAR_LTR SBBidiTypeL
+# define RAQM_TYPE_LTR SBBidiTypeL
+# define RAQM_BIDI_TYPE_WHITESPACE(type) \
+    (type & (SBBidiTypeBN | SBBidiTypeWS | SBBidiTypeLRE | SBBidiTypeRLE | SBBidiTypeLRO | SBBidiTypeRLO | SBBidiTypePDF))
+# define RAQM_NORMAL_BIDI_TYPE_AS_LEVEL(type) \
+    (                                         \
+      ((type) & SBBidiTypeR)                  \
+      ? 1                                     \
+      : 0                                     \
+    )
+# define RAQM_LEVEL_IS_RTL(level) \
+      ((level) & 1)
+  typedef SBLevel raqm_level_t;
+  typedef SBBidiType raqm_par_t;
+  typedef SBBidiType raqm_char_type_t;
+  typedef uint32_t raqm_char_t;
+#else
+# define RAQM_PAR_ON  FRIBIDI_PAR_ON
+# define RAQM_PAR_RTL FRIBIDI_PAR_RTL
+# define RAQM_PAR_LTR FRIBIDI_PAR_LTR
+# define RAQM_TYPE_LTR FRIBIDI_TYPE_LTR
+# define RAQM_BIDI_TYPE_WHITESPACE(type) \
+    (FRIBIDI_IS_EXPLICIT_OR_BN_OR_WS(type))
+# define RAQM_NORMAL_BIDI_TYPE_AS_LEVEL(type) \
+    FRIBIDI_DIR_TO_LEVEL(type)
+# define RAQM_LEVEL_IS_RTL(level) \
+    (FRIBIDI_LEVEL_IS_RTL (level))
+  typedef FriBidiLevel raqm_level_t;
+  typedef FriBidiParType raqm_par_t;
+  typedef FriBidiCharType raqm_char_type_t;
+  typedef FriBidiCharType raqm_char_t;
+#endif
+
```

Comment:
All these types and macros should be has bidi in their names, `raqm_bidi_foo_t` and so on.

Reply:
fixed in 81a755e547b03b1a7cfee8eb71a7afa54c2edc35